### DISCUSSION TITLE
Subworkflow with Predefined workflow definitions

### DISCUSSION THREAD

**Discussion by gau31rav** [Netflix/conductor#2482] [Q&A] [answered]
Hi, I am trying to use sub workflows.
1. Register the child/sub-workflow with output parameters
2. Add a sub-workflow task in parent workflow.
{
	"name": "sub_workflow_task",
	"taskReferenceName": "sub1",
	"type": "SUB_WORKFLOW",
	"inputParameters": {
		"name": "Task 1",
		"task_id": "${CPEWF_TASK_ID}",
		"parent_task_id": null,
		"workflow_id": "${workflow.workflowId}",
		"correlation_id": "${workflow.correlationId}",
		"job_id": "${workflow.input.job_id}",
		"request_message": "${workflow.input.request_message}"
	},
	"subWorkflowParam": {
		"name": "sub_workflow_name"
	},
	"outputParameters": {
		"workflowstatus": "${sub_workflow_task.output.workflowstatus}" "=> This is output of the subworkflow"
	},
	"anythingelse": "value"
}

While registering the parent workflow I am getting validation errors.

 ${sub_workflow_task.output.workflowstatus} is not defined in workflow definition."
 
Is it mandatory to specify workflow definition as part of the sub-workflow task in order to use output parameters of sub-workflow? How can I have sub-workflow output propagated to parent workflow?

**Comment by james-deee** [comment, marked as answer]
I think you need use the `taskReferenceName`. In this case it would be `sub1` instead of using the `name`.

The taskReferenceName is and must be unique across all tasks in a given workflow.

**Reply by gau31rav** [reply]
Thanks, @james-deee for your inputs. Workflow is getting registered successfully.
